### PR TITLE
:hammer: Move chart configs into chart_configs table

### PR DESCRIPTION
**Pull request by Marigold**
## TODO
- [x] Update ORM
- [x] Update all queries with strings `from charts` and `join charts`
- [x] Check all usages of `type` and `slug` in queries

## TODO before merging
- [x] Check that ORM is up to date with MySQL schema

## Testing
- [x] Test `chart-diff`
- [x] Test upserts to MySQL in ETL

### CODE REVIEW

**Comment by Marigold** [comment]
@sophiamersmann this is ready for review (though not tested properly, yet). Let me know when you're ready or when you change schema.

**Comment by sophiamersmann** [comment]
@Marigold the grapher pr also needs to go through review first (and I need to do another round of testing), so it won't be before next week :)

**Comment by sophiamersmann** [comment]
Hey @Marigold, the chart_configs PR on the Grapher side should be ready to merge soon-ish (this or next week). How do things look on the ETL side? (I'd be happy to look over this PR if you want me to)

**Comment by Marigold** [comment]
@sophiamersmann all look good, both schema renames and calling Admin endpoint. Ready whenever you are!

File: etl/grapher_model.py [hunk 1]
```diff
@@ -224,9 +234,29 @@ class ChartRevisions(Base):
     updatedAt: Mapped[Optional[datetime]] = mapped_column(DateTime, init=False)
 
 
+class ChartConfig(Base):
+    __tablename__ = "chart_configs"
+    __table_args__ = (Index("idx_chart_configs_slug", "slug"),)
+
+    id: Mapped[bytes] = mapped_column(BINARY(16), primary_key=True, server_default=text("uuid_to_bin(uuid(),1)"))
+    uuid: Mapped[Optional[str]] = mapped_column(String(36), Computed("(bin_to_uuid(`id`,1))", persisted=False))
```
Comment (sophiamersmann) [inline etl/grapher_model.py:242]: We simplified this and just use a serialised uuid. The full definition of the `chart_configs` table is:

```
CREATE TABLE chart_configs (
    id char(36) NOT NULL PRIMARY KEY,
    patch json NOT NULL,
    full json NOT NULL,
    slug varchar(255) GENERATED ALWAYS AS (JSON_UNQUOTE(JSON_EXTRACT(full, '$.slug'))) STORED,
    createdAt datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
    updatedAt datetime DEFAULT NULL ON UPDATE CURRENT_TIMESTAMP,
    INDEX idx_chart_configs_slug (slug)
)
```

So, removing `uuid` and changing the type of `id` to `String(36)` gets us there 👍🏻

File: etl/grapher_model.py [hunk 2]
```diff
@@ -243,28 +273,37 @@ class Chart(Base):
         ),
         Index("charts_lastEditedByUserId", "lastEditedByUserId"),
         Index("charts_publishedByUserId", "publishedByUserId"),
-        Index("charts_slug", "slug"),
+        Index("configId", "configId", unique=True),
     )
 
     id: Mapped[int] = mapped_column(Integer, primary_key=True, init=False)
-    config: Mapped[dict] = mapped_column(JSON)
+    configId: Mapped[bytes] = mapped_column(BINARY(16))
```
Comment (sophiamersmann) [inline etl/grapher_model.py:280]: Is of a different tyoe, see above 👆🏻

**Pull request merged**